pkg/server/tls/authority/authority.go

```diff
@@ -86,6 +86,8 @@ type DynamicAuthority struct {
 	// watchMutex gates access to the slice of watch channels
 	watchMutex sync.Mutex
 	watches    []chan<- struct{}
+
+	newClient func() (kubernetes.Interface, error) // for testing
```

Comment:
```suggestion
	newClient func(c *rest.Config) (kubernetes.Interface, error) // for testing
```

Which could make the code below a bit simpler - since the function will match `kubernetes.NewForConfig`?